health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/components/UploadDataCustom.js

```
@@ -0,0 +1,1188 @@
+    // data.columns = sortedPropertyNames;
+  }
+
+  function convertIntoSchema(data) {
+    var convertData = { ...data };
+    var properties = {};
+    var required = [];
+    var columns = [];
+
+    for (const propType of ["enumProperties", "numberProperties", "stringProperties"]) {
+      if (convertData?.properties?.[propType] && Array.isArray(convertData?.properties[propType]) && convertData?.properties[propType]?.length > 0) {
+        for (const property of convertData?.properties[propType]) {
```

Biome

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)